ksa_compliance/output_models/e_invoice_output_model.py

```diff
@@ -722,6 +724,24 @@ def get_customer_address_details(self, invoice_id):
     def get_customer_info(self, invoice_id):
         pass
 
+    def compute_invoice_discount_amount(self):
+        discount_amount = self.sales_invoice_doc.discount_amount
+        if self.sales_invoice_doc.apply_discount_on == "Net Total" or discount_amount == 0:
+            self.additional_fields_doc.fatoora_invoice_discount_amount = discount_amount
+            return
+
+        applied_discount_percent = self.sales_invoice_doc.additional_discount_percentage
```

Comment:
I'm assuming this section is meant for the case of applying discount on "Grand Total". We need to explicitly check for that, since that's the aberrant case. You can modify the previous check (net total or discount == 0) to check whether it's not a discount on grand total, which should cover all regular cases.